core/src/main/scala/stryker4s/run/EtaCalculator.scala

```diff
@@ -0,0 +1,40 @@
+package stryker4s.run
+
+import grizzled.slf4j.Logging
+
+import scala.concurrent.duration._
+import language.postfixOps
+
+class EtaCalculator(amountOfMutants: Int) extends Logging {
+  val runResults: Array[Long] = new Array(amountOfMutants)
+  private[this] val stream = Iterator.range(0, amountOfMutants)
+
+  def time[T](fun: => T): (T, String) = {
+    val startTime = System.currentTimeMillis()
+
+    // Run the function
+    val result: T = fun
+
+    // Measure duration and store in array
+    val duration = System.currentTimeMillis() - startTime
+    val runNr = stream.next()
+    runResults(runNr) = duration
+
+    (result, calculateETA(runNr + 1))
+  }
+
+  def calculateETA(currentRun: Int): String = {
+    val from = math.max(0, currentRun - 5) // Base estimate on last 5 runs
+    val average = runResults.slice(from, currentRun).sum / (currentRun - from)
```

Comment:
BTW you can also use `takeRight(n)` to to take the last n elements, or less if the array is smaller.

Reply:
I would use that if this was a Stack that grows in size, but it's an Array. 😁

Reply:
Ah duh. So why not use a mutable.List?

Reply:
Because we know beforehand how long the List is going to be. I believe in that case Array is slightly faster.

Reply:
Hmm, true. Thinking about it. Wouldn't a queue (or whatever) with a max size of 10 (median/average size) be more efficient? Pushing out the oldest element as a new one is added.